src/main/scala/com/target/data_validator/validator/ColumnSumCheck.scala

```diff
@@ -0,0 +1,134 @@
+package com.target.data_validator.validator
+
+import com.target.data_validator.{ValidatorCheckEvent, ValidatorError, VarSubstitution}
+import io.circe._
+import io.circe.generic.semiauto._
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+import org.apache.spark.sql.catalyst.expressions.aggregate.Sum
+import org.apache.spark.sql.types._
+
+case class ColumnSumCheck(
+  column: String,
+  minValue: Option[Json] = None,
+  maxValue: Option[Json] = None,
+  inclusive: Option[Json] = None
+) extends ColumnBased(column, Sum(UnresolvedAttribute(column)).toAggregateExpression()) {
+
+  private val minOrMax: Either[String, Unit] = (minValue, maxValue) match {
+    case (None, None) => Left("'minValue' or 'maxValue' or both must be defined")
+    case _ => Right()
+  }
+
+  private val lowerBound: Either[String, Double] = minValue match {
+    case Some(json) =>
+      if (json.isNumber) { Right(json.asNumber.get.toDouble) }
+      else { Left(s"'minValue' defined but type is not a Number, is: ${json.name}") }
+    case None => Right(Double.MinValue)
+  }
+
+  private val upperBound: Either[String, Double] = maxValue match {
+    case Some(json) =>
+      if (json.isNumber) { Right(json.asNumber.get.toDouble) }
+      else { Left(s"'maxValue' defined but type is not a Number, is: ${json.name}") }
+    case None => Right(Double.MaxValue)
+  }
+
+  private val minLessThanMax: Either[String, Unit] = (lowerBound, upperBound) match {
```

Comment:
Like testing `Option.isEmpty`, testing `Either.isRight` would be more efficient here but I'm less confident in blindly doing stuff with Either without autocomplete in GitHub comments 😆

Reply:
I'll leave this one if you don't mind.  I like that the pattern matching avoids projections in the conditional part and the message.